### PR TITLE
Remove daml on sql artifacts and docs

### PR DESCRIPTION
**Pull request by mziolekda**
CHANGELOG_BEGIN
Stop publishing Daml Driver for PostgreSQL
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Review by cocreature — APPROVED** [review]
Thanks!

**Review by meiersi-da — APPROVED** [review]
Thanks @mziolekda ! Seems like there at least some stragglers left :wink:

It would be great if you could have a look at these to decide for each of them what needs to be done so we don't leave baggage in the codebase.

```
┌─(simon@simon-ThinkPad-v2)-(~/daml)-(mziolek/remove-daml-on-sql-artifacts-dpp-807 ?✔)
└─╼ git grep -i daml-on-sq
ci/cron/src/Docs.hs:        shell_env_ [("DAML_SDK_RELEASE_VERSION", show version)] "bazel build //ledger/daml-on-sql:docs"
ci/cron/src/Docs.hs:        proc_ ["tar", "xzf", "bazel-bin/ledger/daml-on-sql/html.tar.gz", "--strip-components=1", "-C", temp </> show version]
docs/source/ops/connect/helm.rst:      ["-daml-on-sql","-http-json","-oauth2-middleware","-trigger-service",""]
docs/source/support/overview.rst:At the bottom of every Daml Application is a Daml network, a distributed, or possibly centralized persistence infrastructure together with Daml drivers. Daml drivers enable the persistence infrastructure to act as a consensus, messaging, and in some cases persistence layer for Daml Applications. Most Daml drivers will have a public API, but there are no *uniform* public APIs on Daml drivers. This does not harm application portability since applications only interact with Daml networks through the Participant Node. A good example of a public API of a Daml driver is the command line interface of `Daml for Postgres <https://github.com/digital-asset/daml/blob/main/ledger/daml-on-sql/README.rst>`_. It's a public interface, but specific to the Postgres driver.
ledger/daml-on-sql/BUILD.bazel:    name = "daml-on-sql",
ledger/daml-on-sql/BUILD.bazel:    name = "daml-on-sql-test-lib",
ledger/daml-on-sql/BUILD.bazel:        ":daml-on-sql",
ledger/daml-on-sql/BUILD.bazel:    name = "daml-on-sql-binary",
ledger/daml-on-sql/BUILD.bazel:        ":daml-on-sql",
ledger/daml-on-sql/BUILD.bazel:    name = "daml-on-sql-ephemeral-postgresql",
ledger/daml-on-sql/BUILD.bazel:        ":daml-on-sql",
ledger/daml-on-sql/BUILD.bazel:        ":daml-on-sql-test-lib",
ledger/daml-on-sql/BUILD.bazel:    name = "daml-on-sql-tests",
ledger/daml-on-sql/BUILD.bazel:        ":daml-on-sql",
ledger/daml-on-sql/BUILD.bazel:    server = ":daml-on-sql-ephemeral-postgresql",
ledger/daml-on-sql/README.rst:The Community Edition, distributed as a JAR file called `daml-on-sql-<version>.jar`, can be obtained from the `Daml GitHub Releases Page <https://github.com/digital-asset/daml/releases>`__.
ledger/daml-on-sql/README.rst:    $ java -jar daml-on-sql-<version>.jar --ledgerid=test --sql-backend-jdbcurl='jdbc:postgresql://localhost/test?user=fred&password=secret'
ledger/daml-on-sql/README.rst:certificate chain via ``java -jar daml-on-sql-<version>.jar --pem server.pem --crt server.crt``.
ledger/daml-on-sql/README.rst:Sample command to start a server with private key encrypted: ``java -jar daml-on-sql-<version>.jar --pem server.pem.enc --crt server.crt --tls-secrets-url http://localhost:8080``.
ledger/daml-on-sql/README.rst:To start the ledger, run: ``java -jar daml-on-sql-<version>.jar [options]``.
ledger/daml-on-sql/README.rst:``java -jar daml-on-sql-<version>.jar --help``.
ledger/daml-on-sql/kubernetes.yaml:  name: daml-on-sql
ledger/daml-on-sql/kubernetes.yaml:        - name: daml-on-sql
ledger/daml-on-sql/kubernetes.yaml:          image: bazel/ledger/daml-on-sql:daml-on-sql-image
ledger/daml-on-sql/src/main/scala/on/sql/Main.scala:    config.logLevel.foreach(GlobalLogLevel.set("Daml-on-sql"))
ledger/daml-on-sql/src/main/scala/on/sql/package.scala:  private[sql] val Name = LedgerName("Daml-on-SQL")
ledger/metrics/collection/grafana/dashboards/ledger-submissions.json:      "title": "DAML-on-SQL database transaction time",
ledger/metrics/collection/grafana/dashboards/ledger-submissions.json:      "title": "DAML-on-SQL database transaction count",
ledger/metrics/collection/grafana/dashboards/ledger-submissions.json:      "title": "DAML-on-SQL database query time",
ledger/metrics/collection/grafana/dashboards/ledger-submissions.json:      "title": "DAML-on-SQL database query count",
ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/PackageValidationMode.scala:    * Daml-on-SQL, where the validation done in the API server
ledger/sandbox-on-x/README.md:1. Use it as a drop-in-replacement of the implementation of daml-on-sql and sandbox-classic; and thereby enable a massive code cleanup due to removing the direct DB-access code used in daml-on-sql and sandbox-classic.
```

File: ci/cron/src/Docs.hs [hunk 1]
```diff
@@ -224,6 +224,7 @@ damlOnSqlDocOpts :: DocOptions
 damlOnSqlDocOpts = DocOptions
   { s3Subdir = Just "daml-driver-for-postgresql"
   , includedVersion = \v -> v > version "1.8.0-snapshot.20201201.5776.0.4b91f2a6"
+        && v <= version "2.0.0-snapshot.20220209.9212.0.b7fc9f57"
```
[the remark this answers is not in the document]
Comment (cocreature) [reply ci/cron/src/Docs.hs:227]: that doesn’t work. 2.0.0 snapshots are < 2.0.0 and we need to include them or builds will fail for any snapshot between this being merged and 2.0.0 being released.

[the remark this answers is not in the document]
Comment (meiersi-da) [reply ci/cron/src/Docs.hs:227]: Thanks for the clarification; and the careful eye on the PRs. Makes sense.

[the remark this answers is not in the document]
Comment (mziolekda) [reply ci/cron/src/Docs.hs:227]: :+1:

**Comment by mziolekda** [comment]
@meiersi-da Thanks for catching these straddlers. 
daml-on-sql will fly away in a separate PR. This is only about dependencies on daml-on-sql. 
helmcharts will be modified by a separate PR once helmcharts are updated to canton.
docs generation and mentions in sandbox-on-x should stay.

**Pull request merged**